apps/web/src/views/Mev/MevToggle.tsx

```diff
@@ -0,0 +1,118 @@
+import { useTranslation } from '@pancakeswap/localization'
+import {
+  Box,
+  Button,
+  FlexGap,
+  ModalBody,
+  ModalCloseButton,
+  ModalContainer,
+  ModalHeader,
+  ModalTitle,
+  ModalV2,
+  ShieldIcon,
+  Text,
+  Toggle,
+  useModalV2,
+  useTooltip,
+} from '@pancakeswap/uikit'
+import { styled } from 'styled-components'
+import { AddMevRpcButton } from './AddMevRpcButton'
+import { useIsMEVEnabled, useShouldShowMEVToggle } from './hooks'
+import { getImageUrl } from './utils'
+
+export const ToggleWrapper = styled.div`
+  border: 1px solid ${({ theme }) => theme.colors.cardBorder};
+  background-color: ${({ theme }) => theme.colors.tertiary};
+  padding: 16px;
+  display: flex;
+  width: 100%;
+  gap: 4px;
+  border-radius: 16px;
+  align-items: center;
+  justify-content: space-between;
+`
+export const ModalImg = styled.img`
+  width: 258px;
+`
+
+export const MevToggle: React.FC = () => {
+  const { t } = useTranslation()
+  const shouldShowMEVToggle = useShouldShowMEVToggle()
+  const { isMEVEnabled } = useIsMEVEnabled()
+  const { isOpen, onOpen, onDismiss } = useModalV2()
+  const { tooltip, tooltipVisible, targetRef } = useTooltip(
+    t('PancakeSwap MEV Guard protects you from frontrunning and sandwich attacks when Swapping.'),
+    {
+      placement: 'auto',
+      trigger: 'hover',
+    },
+  )
+
+  if (!shouldShowMEVToggle) {
+    return null
+  }
+
+  return (
+    <>
+      <ToggleWrapper>
+        <FlexGap gap="4px">
+          <ShieldIcon width="24px" />
+          <Text>{t('Enable')}</Text>
+          <Text
+            ref={targetRef}
+            style={{ textDecoration: 'underline', textDecorationStyle: 'dotted', cursor: 'pointer' }}
+          >
+            {t('MEV Protect')}
+          </Text>
+          {tooltipVisible && tooltip}
+        </FlexGap>
+        <Toggle scale="md" checked={isMEVEnabled} onClick={onOpen} />
+      </ToggleWrapper>
+      <MevModal isOpen={isOpen} onDismiss={onDismiss} />
+    </>
+  )
+}
+
+export const MevModal: React.FC<{ isOpen: boolean; onDismiss?: () => void; onSuccess?: () => void }> = ({
+  isOpen,
+  onDismiss,
```

Comment:
`onDismiss` we can use this type
import { InjectedModalProps } from '@pancakeswap/uikit'

Reply:
done